### ISSUE TITLE
Add Nextjs config

### ISSUE THREAD

**Issue by thernstig** [ArnaudBarre/eslint-plugin-react-refresh#85]
Vite has its own config. Would it make sense to add one for Next.js? The reason being that Next.js requires this setup:

```js
  {
    plugins: {
      'react-refresh': reactRefresh,
    },
    rules: {
      'react-refresh/only-export-components': [
        'error',
        {
          allowConstantExport: true,
          allowExportNames: ['metadata'],
        },
      ],
    },
  },
```

Next.js latest version has a convention to export these things:

```js
export const metadata = { ... }
export const dynamic = 'force-dynamic'
export const revalidate = 60
export const runtime = 'edge'
export const fetchCache = 'force-no-store'
```

I _believe_ that this plugin is necessary and good even for modern Next.js 15 app router projects, at least when I read https://nextjs.org/docs/architecture/fast-refresh#limitations:

> The file you're editing might have other exports in addition to a React component.

**Comment by ArnaudBarre** [comment]
I'm not a Next.js user, but because it bundle in dev, it's not always mandatory to follow this rule. But back in the day I was using create-react-app, having it fail "sometimes" is very painful and I prefer to have a bit more files but good HMR.

I'm not 100% sure `allowConstantExport` would always with Next, it would be cool if we have an example where exporting an object fail to preserve local state but constant works

**Comment by ArnaudBarre** [comment]
I will try to do some testing next week to decide if it should be enabled or not

**Comment by thernstig** [comment]
@ArnaudBarre it seems `metadata` was not included?
